android/quest/src/main/java/org/smartregister/fhircore/quest/util/PatientUtil.kt

```diff
@@ -0,0 +1,124 @@
+import org.smartregister.fhircore.quest.configuration.view.Properties
+import org.smartregister.fhircore.quest.configuration.view.Property
+import org.smartregister.fhircore.quest.data.patient.model.AdditionalData
+
+suspend fun loadAdditionalData(
+  patientId: String,
+  configurationRegistry: ConfigurationRegistry,
+  fhirEngine: FhirEngine
+): List<AdditionalData> {
+  val result = mutableListOf<AdditionalData>()
+
+  val patientRegisterRowViewConfiguration =
+    configurationRegistry.retrieveConfiguration<PatientRegisterRowViewConfiguration>(
+      configClassification = QuestConfigClassification.PATIENT_REGISTER_ROW
+    )
+
+  patientRegisterRowViewConfiguration.filters?.forEach { filter ->
+    when (filter.resourceType) {
+      Enumerations.ResourceType.CONDITION -> {
+        val conditions =
+          getSearchResults<Condition>(patientId, Condition.SUBJECT, filter, fhirEngine)
+
+        val sortedByDescending = conditions.maxByOrNull { it.recordedDate }
+        sortedByDescending?.category?.forEach { cc ->
+          cc.coding.firstOrNull { c -> c.code == filter.valueCoding!!.code }?.let {
+            val status = sortedByDescending.code?.coding?.firstOrNull()?.display ?: ""
+            result.add(
+              AdditionalData(
+                label = filter.label,
+                value = status,
+                valuePrefix = filter.valuePrefix,
+                properties = propertiesMapping(status, filter)
+              )
+            )
+          }
+        }
+      }
+    }
+  }
+
+  return result
+}
+
+private suspend inline fun <reified T : Resource> getSearchResults(
+  patientId: String,
+  reference: ReferenceClientParam,
+  filter: Filter,
+  fhirEngine: FhirEngine
+): List<T> {
+  return fhirEngine.search {
+    filterByPatient(reference, patientId)
+
+    when (filter.valueType) {
+      Enumerations.DataType.CODEABLECONCEPT -> {
+        filter(TokenClientParam(filter.key), filter.valueCoding!!.asCodeableConcept())
+      }
+      Enumerations.DataType.CODING -> {
+        filter(TokenClientParam(filter.key), filter.valueCoding!!.asCoding())
+      }
+    }
+  }
+}
+
+fun propertiesMapping(value: String, filter: Filter): Properties {
+  return Properties(
+    label = filter.properties?.label,
+    value =
+      Property(
+        color = getColor(value, filter.dynamicColors) ?: filter.properties?.value?.color,
+        textSize = filter.properties?.value?.textSize,
+        fontWeight = filter.properties?.value?.fontWeight
+      )
+  )
+}
+
+fun getColor(value: String, dynamicColors: List<DynamicColor>?): String? {
+  return dynamicColors?.firstOrNull { it.valueEqual == value }?.useColor
+}
+
+fun Search.filterByPatient(reference: ReferenceClientParam, patientId: String) {
+  filter(reference) { this.value = "${ResourceType.Patient.name}/$patientId" }
+}
+
+fun Code.asCoding(): Coding {
+  return Coding(system, code, display)
+}
```

Comment:
Can be reduced to an expression function to improve readability and for conciseness 

```kotlin
fun Code.asCoding(): Coding = Coding(system, code, display)
```
You can further remove the return type as it will be inferred.

Reply:
done